Cargo.toml

```diff
@@ -15,11 +15,11 @@ edition = "2018"
 httparse = "1.3.3"
 async-std = { version = "1.6.0", features = ["unstable"] }
 http-types = "2.0.0"
-pin-project-lite = "0.1.1"
 byte-pool = "0.2.1"
 lazy_static = "1.4.0"
 futures-core = "0.3.1"
 log = "0.4"
+pin-project = "1.0.2"
```

Comment:
in theory, pin-project-lite 0.2 has enum pinning, but i couldn't figure out how to get it to work correctly